### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by haukened**
PR for tab completion.  Github says we can auto merge dev<->dev.

Please review!

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -202,6 +202,107 @@ func populateList() {
...
+    for _, s := range channels {
+        if s.MembersType == keybase.TEAM {
+            // its a team so add the topic name as a possible tab completion
+            firstSlice = append(firstSlice, s.TopicName)
+            firstSlice = append(firstSlice, s.Name)
+        } else {
+            // its a user, so clean the name and append the users name as a possible tab completion
+            firstSlice = append(firstSlice, cleanChannelName(s.Name))
+        }
+    }
```
Comment (Rudi9719) [inline main.go:281]: What happens when you start typing something like "gen"

Comment (haukened) [reply main.go:281]: tab completion will fill `general`
image
image

Comment (haukened) [reply main.go:281]: tabcomplete

Comment (Rudi9719) [reply main.go:281]: It might be better to look into having it so that it ignores channel when team isn't set? Otherwise should someone make a team/user @general there might be some problems lol Or if there are other channels that are mapped to teams/usernames IE there is an @rudi as well as I am in a team with a channel #rudi

Comment (haukened) [reply main.go:281]: ahhh but you missed the reest of the bits.  The flow is this:

1. grab the last word of the input string
2. generate a list of all possible tab completion candidates (even if they are duplicates between teams, channels, and people) that are prefixed by the inputWord
3. determine the longest common prefix (lcp) of all the possible tab completion cantidates
4. insert the longest possible prefix string into the `inputView`

so if your input word is "ru" and your cantidates are `[ "rudi", "rudi9719", and "rudimentary" ]` you would tab-complete only the letters 'di' to make "rudi" and then have to continue typing.

you can see in the gif above i typed "key" and the cantidates were `["keybasefriends", "keybase-pages"]` so initially i got "keybase"  then by typing `f+TAB` i got "keybasefriends"

in short, it behaves exactly the same way that bash completion does!

Comment (Rudi9719) [reply main.go:281]: So what happens if you were to simply hit "return" when it filled `/j gen` to `/j general`? Would it prompt for more info or try to join `@general`

Comment (haukened) [reply main.go:281]: tab completion is only as smart as the user ;-)

Comment (Rudi9719) [reply main.go]: That can be said for a lot of things :P Please modify your PR so that channels are only suggested when a team is already present, to avoid unnecessary errors or messages in the wrong holes. A possible (though I wouldn't hold this against the merge, it's fine to not implement this) idea would also be for channels that are unique (#sec_sector_stigis) maybe those could auto-populate themselves entirely _with_ team name? If not, no big deal
